packages/template-manager/internal/server/template_status.go

```diff
@@ -0,0 +1,34 @@
+package server
+
+import (
+	"context"
+	"fmt"
+	"github.com/e2b-dev/infra/packages/shared/pkg/grpc/template-manager"
+	"go.uber.org/zap"
+)
+
+func (s *serverStore) TemplateBuildStatus(ctx context.Context, in *template_manager.TemplateStatusRequest) (*template_manager.TemplateBuildStatusResponse, error) {
+
+	// todo
+	zap.L().Info("TemplateBuildStatus req", zap.String("buildID", in.BuildID), zap.String("envID", in.TemplateID))
+
+	ctx, ctxSpan := s.tracer.Start(ctx, "template-build-status-request")
+	defer ctxSpan.End()
+
+	logger := s.logger.With(zap.String("buildID", in.BuildID), zap.String("envID", in.TemplateID))
+
+	buildInfo, err := s.buildCache.Get(in.BuildID)
+	if err != nil {
+		return nil, fmt.Errorf("error while getting build info, maybe already expired")
+	}
+
+	if buildInfo.IsFailed() {
```

Comment:
does it make sense to return an error here? you have status error for that right?